### PR TITLE
Fix for CloudPickle Dropping Type Hints

### PR DESCRIPTION
**Pull request by achapkowski**
- Adds fix in if logic so typings are captured for older versions of python.

This resolves issue: https://github.com/cloudpipe/cloudpickle/issues/171

### CODE REVIEW

**Comment by pierreglaser** [comment]
We have tests for annotations, and they used to be skipped for `Python <= 3.6`. Can you remove the `skipif` directive for such tests to see if they passes?

**Comment by achapkowski** [comment]
@pierreglaser updated the tests.

File: tests/cloudpickle_test.py [hunk 1]
```diff
@@ -1588,7 +1588,7 @@ def g():
 
         self.assertEqual(f2.__doc__, f.__doc__)
 
-    @unittest.skipIf(sys.version_info < (3, 7),
+    @unittest.skipIf(sys.version_info < (3, 4),
                      """This syntax won't work on py2 and pickling annotations
                      isn't supported for py36 and below.""")
```
Comment (pierreglaser) [inline tests/cloudpickle_test.py]: The comment should also be updated ("3.6" -> "3.3")

Comment (achapkowski) [reply tests/cloudpickle_test.py]: fixed

**Comment by achapkowski** [comment]
@ogrisel updated the readme.md please recheck.

**Comment by achapkowski** [comment]
@ogrisel @pierreglaser please merge this if no more changes are needed.

**Comment by ogrisel** [comment]
Thanks your contribution @achapkowski!

**Pull request merged**